cpu/atmega_common/periph/uart.c

```diff
@@ -166,45 +166,35 @@ void uart_poweroff(uart_t uart)
 
 static inline void isr_handler(int num)
 {
+    __enter_isr();
     isr_ctx[num].rx_cb(isr_ctx[num].arg, dev[num]->DR);
-
-    if (sched_context_switch_request) {
-        thread_yield();
-    }
+    __exit_isr();
 }
 
 #ifdef UART_0_ISR
 ISR(UART_0_ISR, ISR_BLOCK)
 {
-    __enter_isr();
```

Comment:
I prefer the original design where enter isr is the first line of each ISR

Reply:
Is this just a preference or are there any reasons. As i see there is no benefit in settings the `__in_isr` flag as early as possible, as there is no chance that code could be invoked which could use this information. 
Setting the flag prevents code in the callback to sleep or do other things which schould not be done in interrupts, so setting it before the callback should be sufficient, imho.

I change it as it is like this in all other interrupt handler and thought it would also increase consistency.

Reply:
Personal preference. The consistency helps when reviewing the code and also reduces the risk of missing it if someone uses this periph driver as a template for some other driver for the same CPU.

Reply:
+1

Reply:
@gebart maybe we just have a language problem as consistency translates into two different meanings in german. Do you mean consistency in regards as it stays at it was or consistency as it is similar to other code of the atmegas?
https://github.com/RIOT-OS/RIOT/blob/22aa6b9cc5c0ae95663209a3bdeb13e6b5630e63/cpu/atmega_common/periph/gpio.c#L264-L274
https://github.com/RIOT-OS/RIOT/blob/22aa6b9cc5c0ae95663209a3bdeb13e6b5630e63/cpu/atmega_common/periph/timer.c#L169-L189

Reply:
I guess you are right regarding consistency. I am not very engaged in the atmega development, sorry. 
What I was trying to push for is to have a strict template for the ISR definitions which should be the same across all drivers for the CPU family, which you showed that this is. Consistency makes refactoring the driver code easier in the future. I am happy as long as this design matches the other periph drivers for the atmega.